### PR TITLE
feat(Components/Nodes): better sorting of progress columns [YTFRONT-3801]

### PR DESCRIPTION
**Pull request by ma-efremoff**
https://nda.ya.ru/t/xUb-rkJ5762U47

### CODE REVIEW

File: packages/ui/src/ui/components/ColumnHeader/ColumnHeader.tsx [hunk 1]
```diff
@@ -33,58 +28,134 @@ export interface ColumnHeaderProps<T extends string = string> extends PageCounte
...
+    if (props.options === undefined) {
+        const {column, allowUnordered, withUndefined, title, shortTitle, allowedOrderTypes} = props;
+        return {
+            column,
+            allowUnordered,
+            withUndefined,
+            allowedOrderTypes,
+            ...getNameTitle({column, title, shortTitle}),
+        };
+    } else {
```
Comment (SimbiozizV) [inline packages/ui/src/ui/components/ColumnHeader/ColumnHeader.tsx:86]: "else" is not needed. Above you do "return"

File: packages/ui/src/ui/components/ColumnHeader/ColumnHeader.tsx [hunk 2]
```diff
@@ -100,3 +171,37 @@ export default function ColumnHeader<T extends string = string>(props: ColumnHea
...
+}) {
+    return !options ? null : (
+        <DropdownMenu
+            items={options.map((x) => {
+                const {titleContent} = getNameTitle(x);
+                return {
+                    text: titleContent,
+                    action() {
+                        const {column, allowUnordered, withUndefined, allowedOrderTypes} = x;
+                        const nextOrder = allowedOrderTypes
```
Comment (SimbiozizV) [inline packages/ui/src/ui/components/ColumnHeader/ColumnHeader.tsx:191]: is it worth doing all these calculations if "onSort" may turn out to be "undefined"?

File: packages/ui/src/ui/pages/components/tabs/nodes/Nodes/NodesColumnHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    const dispatch = useDispatch();
+    const sortState = useSelector(getTables)[COMPONENTS_NODES_TABLE_ID];
+    const order = oldSortStateToOrderType(sortState);
+
+    const column = props.options?.find(({column}) => sortState.field === column);
+
+    return (
+        <ColumnHeader
+            {...props}
+            column={props.options ? column?.column ?? props.column : props.column}
```
Comment (SimbiozizV) [inline packages/ui/src/ui/pages/components/tabs/nodes/Nodes/NodesColumnHeader.tsx:26]: double ternary operator. they seem so convenient but this is the path to evil

File: packages/ui/src/ui/store/selectors/tables.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import {RootState} from '../../store/reducers';
+import {initialState} from '../../store/reducers/tables';
+
+export const getTables = (state: RootState) => state?.tables || initialState;
```
Comment (SimbiozizV) [inline packages/ui/src/ui/store/selectors/tables.ts:4]: It's a little confusing. Why not process this in the store? We need to look at in what case "state?.tables" may not exist and why this is not processed in the action

File: packages/ui/src/ui/pages/components/tabs/nodes/Nodes/NodesColumnHeader.tsx (hunk 3 above)
Comment (ma-efremoff) [reply packages/ui/src/ui/pages/components/tabs/nodes/Nodes/NodesColumnHeader.tsx:26]: not double ternary operator

File: packages/ui/src/ui/store/selectors/tables.ts (hunk 4 above)
Comment (ma-efremoff) [reply packages/ui/src/ui/store/selectors/tables.ts:4]: the code was not changed it is just rewritted with typescript

File: packages/ui/src/ui/components/ColumnHeader/ColumnHeader.tsx (hunk 2 above)
Comment (ma-efremoff) [reply packages/ui/src/ui/components/ColumnHeader/ColumnHeader.tsx:191]: `SubColumnSelector` is displayed when `ColumnHeader.props.options` is defined and there is no reason to have `options` without `onSort`.

File: packages/ui/src/ui/components/ColumnHeader/ColumnHeader.tsx (hunk 1 above)
Comment (ma-efremoff) [reply packages/ui/src/ui/components/ColumnHeader/ColumnHeader.tsx:86]: Such isolated `if`-s (without `else`) looks like prevention of some unwanted behaviour below. But here `if/else`-expression is spliting of code-execution for two different types of input props of `HasSortColumn` (it is more like typeguard):
```ts
if (props.options === undefined) {
   // handling of type (ColumnInfo<T> & {options?: undefined})
} else {
  // handling of type 
  // ({
  //        title: string;
  //        shortTitle?: string;
  //        column: T;
  //        options: Array<ColumnInfo<T>>;
  //    } & Partial<Record<keyof Omit<ColumnInfo<T>, 'column' | 'title' | 'shortTitle'>, undefined>>)
}
```
The current form works the same as your suggestion but it is more expressive.

Comment (ma-efremoff) [reply packages/ui/src/ui/components/ColumnHeader/ColumnHeader.tsx]: fixed

File: packages/ui/src/ui/store/selectors/tables.ts (hunk 4 above)
Comment (ma-efremoff) [reply packages/ui/src/ui/store/selectors/tables.ts]: fixed

**Pull request merged**